src/main/java/edu/ucla/library/libcal/handlers/ProxyHandler.java

```diff
@@ -69,19 +79,25 @@ public ProxyHandler(final Vertx aVertx) {
     public void handle(final RoutingContext aContext) {
         final HttpServerResponse response = aContext.response();
         final String path = aContext.request().path();
-
-        final String receivedQuery =
-                path.concat(aContext.request().query() != null ? QUESTION_MARK.concat(aContext.request().query()) : "");
-        myTokenProxy.getBearerToken().compose(token -> {
-            return myApiProxy.getLibCalOutput(token, SLASH.concat(receivedQuery)).onSuccess(apiOutput -> {
-                response.setStatusCode(apiOutput.getInteger(JsonKeys.STATUS_CODE));
-                response.setStatusMessage(apiOutput.getString(JsonKeys.STATUS_MESSAGE));
-                response.putHeader(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON.toString());
-                response.end(apiOutput.getString(JsonKeys.BODY));
+        final String originalClientIP = aContext.request().headers().get(Constants.X_FORWARDED_FOR).split(COMMA)[0];
+        final String[] allowedIPs = myConfig.getString(Config.ALLOWED_IPS).split(COMMA);
```

Comment:
Is this correct? My understanding is that the JavaScript from the Library website will be making API calls to this service, and that code runs on the end users' machines (the IP addresses of which we don't know ahead of time). I expected to see a solution similar to [what we're doing for Hauth](https://github.com/UCLALibrary/hauth/blob/8ba38493a7ef29645d0138b7ed9928f03a2973e0/src/main/java/edu/ucla/library/iiif/auth/handlers/AccessCookieHandler.java#L187-L196), where we configure the app with the _ranges_ that a client IP address must belong to in order to gain access.